bpf/lib/nodeport.h

```diff
@@ -2461,20 +2480,24 @@ lb_handle_health(struct __ctx_buff *ctx __maybe_unused)
 }
 #endif /* ENABLE_HEALTH_CHECK */
 
-static __always_inline int handle_nat_fwd(struct __ctx_buff *ctx)
+static __always_inline int handle_nat_fwd(struct __ctx_buff *ctx, __u8 cluster_id)
 {
 	int ret = CTX_ACT_OK;
 	__u16 proto;
 
+	ctx_store_meta(ctx, CB_CLUSTER_ID_EGRESS, (__u32)cluster_id << 24);
+
```

Comment:
Yeah, I agree with you. Treating cluster_id as `__u8` everywhere is a pain. However, `__u32` `cluster_id` doesn't fit into the mark (including magic number), it consumes more `skb->cb` bandwidth, and every time we downcast it to `__u8` (e.g., when we lookup map), we need to have range check. So, I ended up using `__u8`.

On the other hand, when we want to expand the cluster_id bit width at some point, This may be a pain since we need to expand the bit...

Ok, let me fix them to use `__u32` and do bounds check every time we downcast.

Reply:
Did above!